### PR TITLE
Replace mallinfo by getrusage

### PR DESCRIPTION
**Pull request by murlock**
##### SUMMARY

- Replace mallinfo by getrusage
- Allow sqliterepo_max_rss to be disabled y configuration

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
sqliterepo

##### SDS VERSION
```
5.5.4.dev8
```

### CODE REVIEW

File: sqliterepo/cache.c [hunk 1]
```diff
@@ -118,28 +123,29 @@ struct sqlx_cache_s
...
+			GRID_NOTICE("getrusage failed: %s (%d)", strerror(errno), errno);
+			return FALSE;
+		}
+
+		total = usage.ru_maxrss * 1024;
 		if (total > sqliterepo_max_rss) {
 			GRID_DEBUG("RAM [MiB] used %" G_GINT64_FORMAT" max %" G_GINT64_FORMAT,
 					total / (1024 * 1024),
 					sqliterepo_max_rss / (1024 * 1024));
+			return FALSE;
```
Comment (fvennetier) [inline sqliterepo/cache.c:141]:
```suggestion
			return TRUE;
```

File: sqlx/sqlx_service.c [hunk 2]
```diff
@@ -225,14 +225,18 @@ static void
 _patch_configuration_maxrss(void)
 {
 	const gint64 pre = sqliterepo_max_rss;
-	if (sqliterepo_max_rss <= 0) {
+	if (sqliterepo_max_rss == -1) {
```
Comment (fvennetier) [inline sqlx/sqlx_service.c:228]: Dangerous! If someone puts -2, the code will consider the RAM is always exhausted.
```suggestion
	if (sqliterepo_max_rss < 0) {
```

File: sqliterepo/cache.c (hunk 1 above)
Comment (murlock) [reply sqliterepo/cache.c:141]: :facepalm:

File: sqlx/sqlx_service.c (hunk 2 above)
Comment (murlock) [reply sqlx/sqlx_service.c:228]: Good catch !

**Pull request merged**